src/spatial/detail/ArborX_SpaceFillingCurves.hpp

```diff
@@ -79,6 +61,50 @@ struct Morton64
 namespace Details
 {
 
+template <typename ExecutionSpace, typename Values, typename SpaceFillingCurve,
+          typename Box>
+inline Kokkos::View<std::invoke_result_t<SpaceFillingCurve, Box,
+                                         std::decay_t<decltype(returnCentroid(
+                                             std::declval<Values>()(0)))>> *,
+                    typename Values::memory_space>
+projectOntoSpaceFillingCurve(ExecutionSpace const &space, Values const &values,
+                             SpaceFillingCurve const &curve,
+                             Box const &scene_bounding_box)
+{
+  using Point = std::decay_t<decltype(returnCentroid(values(0)))>;
+  static_assert(GeometryTraits::is_point_v<Point>);
+  static_assert(GeometryTraits::is_box_v<Box>);
+
+  auto const n = values.size();
+
+  using LinearOrderingValueType =
+      std::invoke_result_t<SpaceFillingCurve, Box, Point>;
+  Kokkos::View<LinearOrderingValueType *, typename Values::memory_space>
+      linear_ordering_indices(
+          Kokkos::view_alloc(space, Kokkos::WithoutInitializing,
+                             "ArborX::SpaceFillingCurve::linear_ordering"),
+          n);
+  Kokkos::parallel_for(
+      "ArborX::SpaceFillingCurve::project_onto_space_filling_curve",
+      Kokkos::RangePolicy(space, 0, n), KOKKOS_LAMBDA(int i) {
+        linear_ordering_indices(i) = curve(scene_bounding_box, values(i));
+      });
+
+  return linear_ordering_indices;
+}
+
+template <typename ExecutionSpace, typename Values, typename SpaceFillingCurve,
+          typename Box>
+inline auto computeSpaceFillingCurvePermutation(ExecutionSpace const &space,
+                                                Values const &values,
+                                                SpaceFillingCurve const &curve,
+                                                Box const &scene_bounding_box)
+{
+  auto linear_ordering_indices =
+      projectOntoSpaceFillingCurve(space, values, curve, scene_bounding_box);
+  return sortObjects(space, linear_ordering_indices);
+}
```

Comment:
Saves little code to bundle two distinct operations.
I am not convinced it is a good idea to introduce it.

Reply:
That's fair. It does a bit more than that in that it also reduces the amount of used memory as `linear_ordering_indices` are deallocated automatically once this function returns. This behavior is similar to the original `sortPredicatesAlongSpaceFillingCurve`.